nengo/builder/connection.py

```diff
@@ -76,7 +76,11 @@ def get_targets(conn, eval_points):
     else:
         targets = np.zeros((len(eval_points), conn.size_mid))
         for i, ep in enumerate(eval_points[:, conn.pre_slice]):
-            targets[i] = conn.function(ep)
+            out = conn.function(ep)
+            if out is None:
+                raise BuildError("Building %s: Connection function returned "
```

Comment:
> The memory locations go away if the pre and post of the connection have labels.

Cool, I didn't know connections can have labels! :)

> might be worth making them something that show up in debug mode only?

Yeah, (if it's not a hassle to implement) I'd say so. The error msg that appears when connections have labels is way clearer to me.  I don't think we used `label` keyword once during the summer school tutorials so I am guessing someone new to Nengo will most likely end up with that long error, which is more useful to devs than regular users.